frame: "map"
robot_type: "holonomic"
# robot_type: "differential"
control_frequency: 20
xy_tolerance: 0.01
theta_tolerance: 0.03
can_reverse_drive: 1
replan_threshold: 3
pose_type: 1
replan_frequency_divider: 2

# differential
differential:
  # linear parameters
  lookahead_d: 0.3
  linear_max_vel: 0.5
  linear_acceleration: 1.2
  linear_kp: 2.0
  linear_brake_distance: 0.3
  dacc_vel: 0.3
  linear_margin: 0.00

  # angular parameters
  angular_max_vel: 4.0
  angular_min_vel: 0.1
  # angular_max_vel: 1
  angular_acceleration: 10.0
  # angular_kp: 3
  angular_brake_distance: 0.1
  angular_margin: 0.00

  # spin and dock param
  spin_kp: 2
  dock_kp: 1.2
  spin_and_dock_start_dist: 0.03
  straight_margin_rad: 0.017 # 0.97 degree
  deviate_margin_rad: 0.2
  no_pursuit_y_dis: 0.05
  no_pursuit_rad_tolerance: 3.14 # 30 degree

  # fast_spin_tolerance: 0.03
  # fast_spin_kp: 6.5
  fast_spin_limit: 4.0


# holonomic
holonomic:
  # linear parameters
  lookahead_d: 0.2
  linear_max_vel: 0.7
  linear_acceleration: 0.9
  linear_kp: 2.5
  linear_brake_distance: 0.3
  dacc_vel: 0.3
  linear_margin: 0.00

  # angular parameters
  angular_max_vel: 3
  angular_min_vel: 0.1
  # angular_max_vel: 1
  angular_acceleration: 1.5
  angular_kp: 3
  angular_brake_distance: 0.7
  angular_margin: 0.00

  # spin and dock param
  spin_kp: 2.0
  dock_kp: 1.2
  spin_and_dock_start_dist: 0.03
  deviate_margin_rad: 0.17
  straight_margin_rad: 0.017 # 0.97 degree
  no_pursuit_y_dis: 0.08
  no_pursuit_rad_tolerance: 3.14 # 30 degree

# frame: "map"
# robot_type: "holonomic"
# # robot_type: "differential"
# control_frequency: 20
# xy_tolerance: 0.01
# theta_tolerance: 0.03
# lookahead_d: 0.5
# can_reverse_drive: 1
# replan_threshold: 3
# pose_type: 1
# replan_frequency_divider: 2

# # differential
# differential:
#   # linear parameters
#   linear_max_vel: 0.5
#   linear_acceleration: 0.9
#   linear_kp: 2.5
#   linear_brake_distance: 0.3
#   dacc_vel: 0.3
#   linear_margin: 0.00

#   # angular parameters
#   angular_max_vel: 1.35
#   angular_min_vel: 0.1
#   # angular_max_vel: 1
#   angular_acceleration: 0.8
#   angular_kp: 0.3
#   angular_brake_distance: 0.1
#   angular_margin: 0.00

#   # spin and dock param
#   spin_kp: 1.8
#   dock_kp: 1.2
#   spin_and_dock_start_dist: 0.03
#   straight_margin_rad: 0.017 # 0.97 degree
#   deviate_margin_rad: 0.17
#   no_pursuit_y_dis: 0.08
#   no_pursuit_rad_tolerance: 3.14 # 30 degree


# # holonomic
# holonomic:
#   # linear parameters
#   linear_max_vel: 0.5
#   linear_acceleration: 0.9
#   linear_kp: 2.5
#   linear_brake_distance: 0.3
#   dacc_vel: 0.3
#   linear_margin: 0.00

#   # angular parameters
#   angular_max_vel: 1.35
#   angular_min_vel: 0.1
#   # angular_max_vel: 1
#   angular_acceleration: 0.8
#   angular_kp: 0.3
#   angular_brake_distance: 0.1
#   angular_margin: 0.00

#   # spin and dock param
#   spin_kp: 2.0
#   dock_kp: 1.2
#   spin_and_dock_start_dist: 0.03
#   deviate_margin_rad: 0.17
#   straight_margin_rad: 0.017 # 0.97 degree
#   no_pursuit_y_dis: 0.08
#   no_pursuit_rad_tolerance: 3.14 # 30 degree
  fast_spin_tolerance: 0.03
  fast_spin_kp: 3.5
  fast_spin_limit: 2.5